script/bootstrap: update openssl before stage3

jepio opened the pull request:
# script/bootstrap: update openssl before stage3

Depclean failures in stage3 have been tracked down to stage1 containing openssl-1.1, and stage3 packages being built against that (dependencies are soft enough that they are not ordered against openssl-3 update in stage3). Add a workaround: explicitly build openssl-3 in stage2, right before stage3.

## How to use

`./boostrap_sdk`.

## Testing done

CI will run.

dongsupark commented:
Looks good if CI passed.

But can you please add an `einfo` message for openssl [here](https://github.com/flatcar-linux/portage-stable/blob/e20d2e4d478ef76de7dba0a04e39414914586d19/scripts/bootstrap.sh#L301)?

jepio commented:
I rebased to resolve merge conflicts with the curl addition and added the einfo call. I wanted to hold off on merging to observe some more, but this still happens occasionally in testing so merging it now.

dongsupark commented:
Thanks for merging it.

Site note: When we have some time, we should fix the same issues in upstream Gentoo.
Including this PR and https://github.com/flatcar-linux/portage-stable/pull/240/commits/876ddceb688a4bb96b59a131ccecdfbe347dc406 .

tormath1 commented:
@dongsupark great idea. Is there an existing bug report to ease the reviewing process ?

dongsupark commented:
Not sure. But there is a similar issue https://bugs.gentoo.org/817914 as the curl ssl issue.

jepio commented:
The openssl issue is self-inflicted because our stage1 does not use the same ebuilds as stage2-3, so that one does not need to go upstream.

It would make sense to push the curl fix upstream though. Gentoo is probaby not hitting it because all sources needed for bootstrap are available on the mirror?

dongsupark commented:
> Gentoo is probaby not hitting it because all sources needed for bootstrap are available on the mirror?

Right.
While flatcar-linux/baselayout fetches from GIthub, the Gentoo's baselayout does not touch Github. That's why the issue has been hidden in Gentoo.
https://gitweb.gentoo.org/repo/gentoo.git/tree/sys-apps/baselayout/baselayout-9999.ebuild#n14
